tvb/simulator/models/epileptorcodim3.py

```diff
@@ -0,0 +1,720 @@
+        if self.N == 1:
+            xs = (mu1 / 2.0 + numpy.sqrt(
+                mu1 ** 2 / 4.0 - mu2 ** 3 / 27.0 + 0 * 1j)) ** (1.0 / 3.0) + (mu1 / 2.0 - numpy.sqrt(
+                mu1 ** 2 / 4.0 - mu2 ** 3 / 27.0 + 0 * 1j)) ** (1.0 / 3.0)
+        elif self.N == 2:
+            xs = -1.0 / 2.0 * (1.0 - 1j * 3 ** (1.0 / 2.0)) * (mu1 / 2.0 + numpy.sqrt(
+                mu1 ** 2 / 4.0 - mu2 ** 3 / 27.0 + 0 * 1j)) ** (1.0 / 3.0) - 1.0 / 2.0 * (
+                1.0 + 1j * 3 ** (1.0 / 2.0)) * (mu1 / 2.0 - numpy.sqrt(mu1 ** 2 / 4.0 - mu2 ** 3 / 27.0 + 0 * 1j)) ** (
+                1.0 / 3.0)
+        elif self.N == 3:
+            xs = -1.0 / 2.0 * (1.0 + 1j * 3 ** (1.0 / 2.0)) * (mu1 / 2.0 + numpy.sqrt(
+                mu1 ** 2 / 4.0 - mu2 ** 3 / 27.0 + 0 * 1j)) ** (1.0 / 3.0) - 1.0 / 2.0 * (
+                1.0 - 1j * 3 ** (1.0 / 2.0)) * (mu1 / 2.0 - numpy.sqrt(mu1 ** 2 / 4.0 - mu2 ** 3 / 27.0 + 0 * 1j)) ** (
+                1.0 / 3.0)
+        xs = numpy.real(xs)
+
+        xdot = -y
+        ydot = x ** 3 - mu2 * x - mu1 - y * (nu + self.b * x + x ** 2)
+        if self.modification:
+            zdot = -self.c * (
+                numpy.sqrt((x - xs) ** 2 + y ** 2) - self.dstar + 0.1 * (z - 0.5) ** 7 + self.Ks * coupling[0, :])
+        else:
+            zdot = -self.c * (numpy.sqrt(
+                (x - xs) ** 2 + y ** 2) - self.dstar + self.Ks * coupling[0, :])
+
+        derivative = numpy.array([xdot, ydot, zdot])
+        return derivative
+
+    def update_derived_parameters(self):
+        r"""
+        The equations were taken from [Saggioetal_2017]
+        cf. Eqn. (7), page 17
+
+        Here we parametrize the great arc which lies on a sphere of radius R
+        between the points A and B, which are given by:
+
+            .. math::
+                A &= \begin{pmatrix}\mu_{2,start} & -\mu_{1,start} & \nu_{start} \end{pmatrix} \\
+                B &= \begin{pmatrix}\mu_{2,stop} & -\mu_{1,stop} & \nu_{stop} \end{pmatrix}
+
+        Then we parametrize this great arc with z as parameter by :math:`R(E \cos z + F \sin z)`
+            where the unit vectors E and F are given by:
+
+            .. math::
+                E &= A/\|A\| \\
+                F &= ((A \times B) \times A)/\|(A \times B) \times A\|
+        """
+
+        A = numpy.array(
+            [self.mu2_start[0], -self.mu1_start[0], self.nu_start[0]])
+        B = numpy.array([self.mu2_stop[0], -self.mu1_stop[0], self.nu_stop[0]])
+
+        self.E = A / numpy.linalg.norm(A)
+        self.F = numpy.cross(numpy.cross(A, B), A)
+        self.F = self.F / numpy.linalg.norm(self.F)
+
+    def dfun(self, state_variables, coupling, local_coupling=0.0):
+        """"The dfun using numba for speed"""
+        state_variables_ = state_variables.reshape(state_variables.shape[:-1]).T
+        coupling_ = coupling.reshape(coupling.shape[:-1]).T
+        derivative = _numba_dfun(state_variables_, coupling_, self.E[0], self.E[1], self.E[2], self.F[0], self.F[1],
```

Comment:
@LenSpek can you set a breakpoint here and double check that the types of those non-float parameters being passed to the numba dfun match or if not, cast their values correclty?

Reply:
I double-checked and the variables in question (self.modification, self.N) have dtype int32.

Reply:
is that the case for defaults as well (when you don't specify any model parameters)?  this test runs your model as if you made a simulation without specifying any parameters, only defaults